### PR TITLE
`system_malloc` OK

### PR DESCRIPTION
**Pull request by AngheloAlf**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

This was a disappointing file to do :c

### CODE REVIEW

File: tools/disasm/files.txt [hunk 1]
```diff
@@ -255,7 +255,7 @@
             0x80096C30 : "loadfragment2",
             0x80096C40 : "stackcheck",
             0x80096C50 : "gfxprint",
-            0x80097500 : "system_malloc",
+            0x80097500 : "",
```
Comment (isghj5) [inline tools/disasm/files.txt:258]: Uh.. is this an error?

File: src/boot_O2/system_malloc.c [hunk 2]
```diff
@@ -0,0 +1,50 @@
+#include "global.h"
+
+Arena gSystemArena;
+
+void* SystemArena_Alloc(u32 size) {
```
Comment (Thar0) [inline src/boot_O2/system_malloc.c:5]:
```suggestion
void* SystemArena_Malloc(u32 size) {
```
I think we should sync with OoT names fully for this file.

File: src/boot_O2/system_malloc.c [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+void SystemArena_InitArena(void* start, u32 size) {
+    __osMallocInit(&gSystemArena, start, size);
+}
+
+void SystemArena_Cleanup(void) {
+    __osMallocCleanup(&gSystemArena);
+}
+
+void SystemArena_IsInitialized(void) {
+    __osMallocIsInitalized(&gSystemArena);
```
Comment (Thar0) [inline src/boot_O2/system_malloc.c:49]:
```suggestion
u8 SystemArena_IsInitialized(void) {
    return __osMallocIsInitalized(&gSystemArena);
```

File: tools/disasm/files.txt (hunk 1 above)
Comment (AngheloAlf) [reply tools/disasm/files.txt:258]: system_malloc doesn't has any data, and the data in that split seems to be referenced by another file.

File: src/boot_O2/system_malloc.c (hunk 3 above)
Comment (AngheloAlf) [reply src/boot_O2/system_malloc.c]: Yeah, I already noted this and fixed it, but forgot to push :sweat_smile:

**Review by Kenix3 — APPROVED** [review]

**Comment by Kenix3** [comment]
Merge is pending +2 reviewer approvals.

**Review by Thar0 — APPROVED** [review]

File: tools/disasm/files.txt (hunk 1 above)
Comment (EllipticEllipsis) [reply tools/disasm/files.txt:258]: I'd rather we figured this out properly first: I don't know what an empty split name does to the disasm.

Comment (Thar0) [reply tools/disasm/files.txt:258]: An empty name is acceptable, it's just not obvious what .text section it's associated with if it's left empty.

Comment (EllipticEllipsis) [reply tools/disasm/files.txt:258]:
```suggestion
            0x80097500 : "boot_80086760",
```

Comment (AngheloAlf) [reply tools/disasm/files.txt:258]: Are you sure that's the .text file corresponding to this .data?

Comment (EllipticEllipsis) [reply tools/disasm/files.txt:258]: Pretty sure:
image

File: tools/disasm/files.txt [hunk 4]
```diff
@@ -256,7 +256,7 @@
             0x80096C30 : "loadfragment2",
             0x80096C40 : "stackcheck",
             0x80096C50 : "gfxprint",
-            0x80097500 : "system_malloc",
+            0x80097500 : "boot_800862E0",
```
Comment (engineer124) [inline tools/disasm/files.txt:259]: Is there a reason this was unnamed?

**Pull request merged**